src/Button/native/Button.js

```diff
@@ -1,49 +1,43 @@
 import React from 'react';
-import { TouchableOpacity, TouchableNativeFeedback, Platform } from 'react-native';
 import PropTypes from 'prop-types';
 import styled from 'styled-components';
 import theme from '../../theme';
+import Text from '../../Text/native';
 
-const ButtonView = styled.View`
+const styles = {
+  buttonColor(props) {
+    switch (props.kind) {
+      case 'filled': return props.theme.color[props.color];
+      default: return props.theme.color.white;
+    }
+  },
```

Comment:
The inherit problem with switch cases is that you can never return a value based on two conditions, in this particular situation you're comparing a value to a single constant string. 

If tomorrow you need to compare two values, you'll need to replace the entire switch case with an if-else-if ladder.

```
if (props.kind === 'filled' && props.size === 'small')
```